connectors/src/connectors/intercom/temporal/sync_help_center.ts

```diff
@@ -367,16 +369,56 @@ export async function upsertArticle({
       updatedAt: updatedAtDate,
     });
 
+    const documentId = getHelpCenterArticleInternalId(connectorId, article.id);
+
     // Parents in the Core datasource should map the internal ids that we use in the permission modal
-    // Parents of an article are all the collections above it and the help center + the article itself
-    const parentsInternalsIds = article.parent_ids.map((id) =>
-      getHelpCenterCollectionInternalId(connectorId, id.toString())
+    // Order is important: We want the id of the article, then all parents collection in order, then the help center
+    const articleinternalId = documentId;
+    const parentCollectioninternalId = getHelpCenterCollectionInternalId(
+      connectorId,
+      parentCollectionId
     );
-    parentsInternalsIds.push(
-      getHelpCenterInternalId(connectorId, helpCenterId)
+    let grandParentCollectioninternalId = null;
```

Comment:
Suggestion to improve code readability, a new function something like 👇 . I guess you'll also need it for the backfill anyway.

```
// Parents in the Core datasource should map the internal ids that we use in the permission modal
// Order is important: We want the id of the article, then all parents collection in order, then the help center
async function getParentIdsForArticle(documentId, connectorId, parentCollectionId, helpCenterId) {
  // Initialize the internal IDs array with the article ID.
  const internalIds = [documentId];

  // Add the parent collection ID.
  internalIds.push(getHelpCenterCollectionInternalId(connectorId, parentCollectionId));

  // Fetch and add any grandparent collection IDs.
  let currentParentId = parentCollectionId;
  
  // There's max 2-levels on Intercom.
  for (let i = 0; i < 2; i++) { 
    const currentParent = await IntercomCollection.findOne({
      where: {
        connectorId,
        collectionId: currentParentId,
      },
    });

    if (currentParent && currentParent.parentId) {
      currentParentId = currentParent.parentId;
      internalIds.push(getHelpCenterCollectionInternalId(connectorId, currentParentId));
    } else {
      break;
    }
  }

  // Add the help center internal ID
  internalIds.push(getHelpCenterInternalId(connectorId, helpCenterId));

  return internalIds;
}
```

Reply:
I had something like that initially and those this is less readable 😅